src/algorand/algorandTypes.ts

```diff
@@ -155,7 +155,7 @@ export const asMaybeCustomFee = asMaybe(
 
 export const asMaybeContractAddressLocation = asMaybe(
   asObject({
-    contractAddress: asString
+    contractAddress: asIntegerString
```

Comment:
Now that you are doing this, you can delete the `isNaN` check in AlgorandTools.ts.